### PR TITLE
chore: dispatch loading failed event

### PR DESCRIPTION
**Pull request by mrehan27**
part of [MBL-480](https://linear.app/customerio/issue/MBL-480/update-in-app-messaging-module-using-state-management-in-ios)

### Changes

- Removed `EngineRoute` class as it seemed to be an unnecessary wrapper
- Dispatched `messageLoadingFailed` action when an in-app message times out
- Replaced `engineAction.error` with `engineAction.messageLoadingFailed` to simplify and align with Android
- Cleaned up and simplified the `Message` class
- Updated `InAppMessageReducer` to handle `messageLoadingFailed` action

### CODE REVIEW

File: Sources/MessagingInApp/Gist/EngineWeb/EngineWeb.swift [hunk 1]
```diff
@@ -21,6 +21,8 @@ protocol EngineWebInstance: AutoMockable {
 
 public class EngineWeb: NSObject, EngineWebInstance {
     private let logger: Logger = DIGraphShared.shared.logger
+    private let inAppMessageManager: InAppMessageManager = DIGraphShared.shared.inAppMessageManager
+    private let _currentMessage: Message?
```
Comment (Shahroz16) [inline Sources/MessagingInApp/Gist/EngineWeb/EngineWeb.swift:25]: Can this be part of the constructor? The usage of EngineWeb only seems to be from MessageManager and it get initialized with Message.

File: Sources/MessagingInApp/Gist/Managers/MessageManager.swift [hunk 2]
```diff
@@ -40,7 +40,7 @@ class MessageManager: EngineWebDelegate {
             instanceId: message.instanceId,
             endpoint: state.environment.networkSettings.engineAPI,
             messageId: message.messageId,
-            properties: message.toEngineRoute().properties
+            properties: message.properties.mapValues { AnyEncodable($0) }
```
[the remark this answers is not in the document]
Comment (mrehan27) [reply Sources/MessagingInApp/Gist/Managers/MessageManager.swift:43]: I don't think `AnyEncodable` directly throws anything. We've been doing the same when building `EngineRoute` class previously, so nothing really has changed here.

Also, I tried calling it like `try? AnyEncodable($0)` and could see the following warning:

```
No calls to throwing functions occur within 'try' expression
```

File: Sources/MessagingInApp/Gist/EngineWeb/EngineWeb.swift (hunk 1 above)
Comment (mrehan27) [reply Sources/MessagingInApp/Gist/EngineWeb/EngineWeb.swift]: Updated and made it non-null

**Review by Shahroz16 — APPROVED** [review]
Looks good.

**Pull request merged**